ingeniamotion/pdo.py

```diff
@@ -18,6 +21,130 @@
     from ingeniamotion.motion_controller import MotionController
 
 
+class PDOPoller:
+    """Poll register values using PDOs"""
+
+    def __init__(
+        self,
+        mc: "MotionController",
+        servo: str,
+        refresh_time: float,
+        buffer_size: int,
+    ) -> None:
+        """Constructor.
+
+        Args:
+            mc: MotionController instance
+            servo: drive alias.
+            refresh_time: PDO values refresh time.
+            buffer_size: Maximum number of register readings to store.
+
+        """
+        super().__init__()
+        self.__mc = mc
+        self.__servo = servo
+        self.__refresh_time = refresh_time
+        self.__buffer_size = buffer_size
+        self.__num_channels: int = 0
+        self.__buffer: Dict[int, Deque[Union[int, float]]] = defaultdict(partial(deque, maxlen=self.__buffer_size))  # type: ignore
+        self.__timestamps: Deque[float] = deque(maxlen=self.__buffer_size)
+        self.__start_time: Optional[float] = None
+        self.__tpdo_map: TPDOMap = self.__mc.capture.pdo.create_empty_tpdo_map()
+        self.__rpdo_map: RPDOMap = self.__mc.capture.pdo.create_empty_rpdo_map()
+        self.__fill_rpdo_map()
+
+    def start(self) -> None:
+        """Start the poller"""
+        self.__mc.capture.pdo.set_pdo_maps_to_slave(
+            self.__rpdo_map, self.__tpdo_map, servo=self.__servo
+        )
+        self.__mc.capture.pdo.subscribe_to_receive_process_data(self._new_data_available)
+        self.__start_time = time.time()
+        self.__mc.capture.pdo.start_pdos(refresh_rate=self.__refresh_time)
+
+    def stop(self) -> None:
+        """Stop the poller"""
+        self.__mc.capture.pdo.stop_pdos()
+        self.__mc.capture.pdo.unsubscribe_to_receive_process_data(self._new_data_available)
+        self.__mc.capture.pdo.remove_rpdo_map(self.__servo, self.__rpdo_map)
+        self.__mc.capture.pdo.remove_tpdo_map(self.__servo, self.__tpdo_map)
+
+    @property
+    def data(self) -> Tuple[Deque[float], Dict[int, Deque[Union[int, float]]]]:
+        """
+        Get the poller data. After the data is retrieved, the data buffers are cleared.
+
+        Returns:
+            A list with the readings timestamps and values.
```

Comment:
The returns should be explained better in the docstring.

Reply:
Done